scripts/approximations/rational.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+
+def construct_vandermonde_matrix(x_list: list, y_list: list) -> list[list]:
+    """ Constructs a Vandermonde matrix for a rational approximation.
+        from the list of x and y values given.
+        len(x_list) * len(x_list)
+        x_list is the list of all x values to construct the matrix from.
+        V = [1 x_0 x_0^2 ... x_0^{n-1} - y_0*x_0 - y_0*x_0^2 ... - y_0*x_0^{n-1}]
+            [1 x_1 x_2^1 ... x_1^{n-1} - y_1*x_1 - y_1*x_1^2 ... - y_1*x_1^{n-1}]
+            ...
+            [1 x_n x_n^2 ... x_n^{n-1} - y_n*x_n - y_n*x_n^2 ... - y_n*x_n^{n-1}]
+    """
+    num_terms = (len(x_list) + 1) // 2
+
+    matrix = []
+
+    for i in range(num_terms * 2 - 1):
+        row = []
+        for j in range(num_terms):
+            row.append(x_list[i]**j)
+
+        for j in range(num_terms):
+            # denominator terms
+            if j > 0:
+                row.append(-1 * x_list[i]**j * y_list[i])
+
+        matrix.append(row)
+
+    return matrix
+
+def solve(x: np.ndarray, coefficients_numerator: list, coefficients_denominator: list) -> np.ndarray:
```

Comment:
This should probably be renamed `evaluate`, were not solving the system of equations in this fn

Reply:
EDIT: Same as comment as what I made for polynomial, I don't think this works.

Reply:
Please see: https://github.com/osmosis-labs/osmosis/pull/3620#discussion_r1041255606

We are not solving but evaluating here. This is needed to compute the y coordinates for the final plot. We do this once the functions are already approximated and we have the coefficients

Reply:
Resolving based on discussion in: https://github.com/osmosis-labs/osmosis/pull/3620#discussion_r1041362613